api/src/main/java/openconsensus/metrics/DoubleCumulative.java

```diff
@@ -0,0 +1,147 @@
+ *                       "Processed jobs", "1", labelKeys);
+ *
+ *   // It is recommended to keep a reference of a point for manual operations.
+ *   DoublePoint inboundPoint = cumulative.getOrCreateTimeSeries(labelValues);
+ *
+ *   void doSomeWork() {
+ *      // Your code here.
+ *      inboundPoint.set(15);
+ *   }
+ *
+ * }
+ * }</pre>
+ *
+ * @since 0.1.0
+ */
+@ThreadSafe
+public abstract class DoubleCumulative {
+
+  /**
+   * Creates a {@code TimeSeries} and returns a {@code DoublePoint} if the specified {@code
+   * labelValues} is not already associated with this cumulative, else returns an existing {@code
+   * DoublePoint}.
+   *
+   * <p>It is recommended to keep a reference to the DoublePoint instead of always calling this
+   * method for manual operations.
+   *
+   * @param labelValues the list of label values. The number of label values must be the same to
+   *     that of the label keys passed to {@link MetricRegistry#addDoubleCumulative}.
+   * @return a {@code DoublePoint} the value of single cumulative.
+   * @throws NullPointerException if {@code labelValues} is null OR any element of {@code
+   *     labelValues} is null.
+   * @throws IllegalArgumentException if number of {@code labelValues}s are not equal to the label
+   *     keys.
+   * @since 0.1.0
+   */
+  public abstract DoublePoint getOrCreateTimeSeries(List<LabelValue> labelValues);
+
+  /**
+   * Returns a {@code DoublePoint} for a cumulative with all labels not set, or default labels.
+   *
+   * @return a {@code DoublePoint} for a cumulative with all labels not set, or default labels.
+   * @since 0.1.0
+   */
+  public abstract DoublePoint getDefaultTimeSeries();
+
+  /**
+   * Removes the {@code TimeSeries} from the cumulative metric, if it is present. i.e. references to
+   * previous {@code DoublePoint} objects are invalid (not part of the metric).
+   *
+   * @param labelValues the list of label values.
+   * @throws NullPointerException if {@code labelValues} is null or any element of {@code
+   *     labelValues} is null.
+   * @since 0.1.0
+   */
+  public abstract void removeTimeSeries(List<LabelValue> labelValues);
+
+  /**
+   * Removes all {@code TimeSeries} from the cumulative metric. i.e. references to all previous
+   * {@code DoublePoint} objects are invalid (not part of the metric).
+   *
+   * @since 0.1.0
+   */
+  public abstract void clear();
+
+  /**
+   * The value of a single point in the Cumulative.TimeSeries.
+   *
+   * @since 0.1.0
+   */
+  public abstract static class DoublePoint {
+
+    /**
+     * Adds the given value to the current value. The values cannot be negative.
+     *
+     * @param delta the value to add
+     * @since 0.1.0
+     */
+    public abstract void add(double delta);
+
+    /**
+     * Sets the given value.
+     *
+     * @param val the new value.
+     * @since 0.1.0
+     */
+    public abstract void set(double val);
```

Comment:
We should not allow set on a always increasing metric (cumulative or counter).

Reply:
`Set` is very useful when you have already-aggregated cumulative metrics, such as CPU stat from /proc. Without this API users will be forced to keep track of previous values and calculate deltas (for example https://github.com/census-instrumentation/opencensus-service/blob/master/receiver/vmmetricsreceiver/vm_metrics_collector.go#L96-L101). `Set` will be a noop if the given value is less than the current one.

Go also has this: https://github.com/census-instrumentation/opencensus-go/blob/dev/metric/cumulative.go#L64.

Reply:
Why not using the Derived version for already aggregated metrics?

Reply:
Then we're forcing users to define callbacks. I think we need to support both callback and pushing for aggregated metrics.

Reply:
Not sure I understand the need of set, because for the CPU definitely defining a callback is better instead of defining your own timer.

Reply:
Chatted offline, we cannot think of a good example of using `Set`. Removed this API for now.